net/can/can_recvmsg.c

```diff
@@ -622,7 +623,14 @@ ssize_t can_recvmsg(FAR struct socket *psock, FAR struct msghdr *msg,
    * socket has been disconnected.
    */
 
+  /* The following operations must be performed with interrupts
+   * disabled because iob_tryadd_queue() may be called from an interrupt
```

Comment:
iob_tryalloc_qentry already hold critical section, why we need hold here again?

Reply:
iob_tryalloc_qentry is already in CAN interrupt ,but iob_remove_queue in can_recvmsg does not  hold critical section

Reply:
we should fix IOB function instead, so not only can socket, but also other module which invoke IOB function can work too.